src/rgw/driver/rados/rgw_rest_user.cc

```diff
@@ -237,6 +243,9 @@ void RGWOp_User_Create::execute(optional_yield y)
   if (s->info.args.exists("system"))
     op_state.set_system(system);
 
+  if (s->info.args.exists("account-root"))
+    op_state.set_account_root(account_root);
```

Comment:
Documentation states that:
> Each account is managed by an account root user.

Should there be a check to see if there already is a root user on the account? Currently clients can create multiple root users for a single account, but documentation implies there should only be one root user.

Reply:
yeah, i just didn't think it was worth the extra complexity to limit accounts to a single root user

the docs discuss what i expect to be the most common use case, where an admin creates an account and a single root user on behalf of an end-user

the "Migrate an existing User into an Account" section mentions that existing users could be migrated to account-root users, and i don't see a reason to prevent multiple such users from being added to the same account

what do you think?

Reply:
For ceph I can't think of any concerns other then that its inconsistent with AWS accounts only having one root user per account. Other than that I see no downsides.

Only thing I would recommend is making the documentation a little more clear that an account could have one or more root users. When I first noticed the behavior I wasn't sure if it was intended. Thanks Casey!